### PR TITLE
holoinstall: Don't wipe the whole disk

### PR DESCRIPTION
**Pull request by PIPIPIG233666**
* let's make dual booting easy and possible

 * basically this change gets the start of any given free space and applies some simple 1st grade addition to get the end

### CODE REVIEW

**Comment by PIPIPIG233666** [comment]
Tested on my hardware with the following:<details>
<summary>Config</summary>

> nvme1n1                                  465.8G disk
> ├─nvme1n1p1 Microsoft reserved partition    16M part
> ├─nvme1n1p2 Basic data partition         119.2G part
> ├─nvme1n1p3 Basic data partition             8G part
> ├─nvme1n1p4 Basic data partition           128G part
> ├─nvme1n1p5 CLR_MNT_/home                119.2G part
> ├─nvme1n1p6                               14.9G part
> **├─nvme1n1p7 primary                        244M part**
> **├─nvme1n1p8 primary                       22.1G part**
> **└─nvme1n1p9 primary                       54.1G part**
</details>

To start, get a (GPT, of course) drive with some free space, just do whatever was done before the change, and the script will install holo along with the existing partition schemes.

**Comment by theVakhovskeIsTaken** [comment]
That's cool, but, what if user wants to fully wipe his disk?

**Comment by PIPIPIG233666** [comment]
> That's cool, but, what if user wants to fully wipe his disk?

We can make both optional 😃

**Comment by wynn1212** [comment]
Well. you forget about adding the drive confirmation screen. Although it looks like won't erase the whole disk. But adding drive confirmation screen back fell better safe than sorry.

**Comment by theVakhovskeIsTaken** [comment]
Alright, looks like the thing looks clear, but supplying installer with ISO will be deprecated on next release, and i'll suggest everyone to push their installer PRs to: https://github.com/theVakhovskeIsTaken/holoiso-install-script

**Comment by PIPIPIG233666** [comment]
> Alright, looks like the thing looks clear, but supplying installer with ISO will be deprecated on next release, and i'll suggest everyone to push their installer PRs to: [theVakhovskeIsTaken/holoiso-install-script](https://github.com/theVakhovskeIsTaken/holoiso-install-script)

Will do when I reboot into linux some time later. 😆 back to windows to play some XGP games

File: airootfs/usr/local/bin/holoinstall [hunk 1]
```diff
@@ -37,56 +37,91 @@ fi
...
+	read ANS
+
+	if [[ ! ($ANS = 'n' || $ANS = 'N') ]]; then
+		echo "\nWiping partitions..."
+		sfdisk --delete ${DEVICE}
+		wipefs -a ${DEVICE}
+		echo "\nCreating new gpt partitions..."
+		parted ${DEVICE} mklabel gpt
+		destructive=true
+	fi
```
Comment (wynn1212) [inline airootfs/usr/local/bin/holoinstall]: Just wondering. What if user accidentally type other than `n` or `N`?

The whole disk got erased.

**Review by wynn1212 — COMMENTED** [review]
There might be the problem in drive confirmation selection screen.

File: airootfs/usr/local/bin/holoinstall (hunk 1 above)
Comment (PIPIPIG233666) [reply airootfs/usr/local/bin/holoinstall]: Done

**Comment by PIPIPIG233666** [comment]
https://github.com/theVakhovskeIsTaken/holoiso-install-script/pull/1

**Pull request closed without merging**